### PR TITLE
1796 add any field handling for search api

### PR DESCRIPTION
**Pull request by gergely-ujvari**
Rewritten version of #1846 

Less things have changed because now we do not have store dependency

**Lost functionality**
- The since search keyword is no longer working. For that to work we have to further enhance our search API

### CODE REVIEW

**Comment by gergely-ujvari** [comment]
An important comment: The since keyword support was lost when we left using the websocket api, so this PR is not responsible for that.

File: h/static/scripts/searchfilters.coffee [hunk 1]
```diff
@@ -43,6 +43,29 @@ class SearchFilter
...
+
+    addToObj = (key, data) ->
+      if obj[key]?
+        obj[key].push data
+      else
+        obj[key] = [data]
+
+    if searchtext
+      filter = null
+      data = null
```
Comment (aron) [inline h/static/scripts/searchfilters.coffee]: These two lines are redundant.

Comment (gergely-ujvari) [inline h/static/scripts/searchfilters.coffee]: On 2015-01-19 11:39, Aron Carroll wrote:
>
> These two lines are redundant.
>
Yeah, removing them.

**Comment by gergely-ujvari** [comment]
Okay, @aron   reviewed the frontend, @Treora reviewed the backend, as soon as travis finished, merging.

**Pull request merged**